presentation/src/main/java/com/nexters/boolti/presentation/screen/ticket/detail/TicketDetailScreen.kt

```diff
@@ -175,8 +196,20 @@ private fun TicketDetailScreen(
         }
     }
 
+    LaunchedEffect(pagerState.currentPage) {
+        snapshotFlow { pagerState }
+            .map { it.currentPage }
+            .onEach(viewModel::syncCurrentPage)
+            .launchIn(this)
+    }
```

Comment:
[질문] 내가 snapshotFlow를 몰라서... 이게 snapshotFlow에 block으로 전달한 값의 변경을 추적하여 로직을 실행하는 거지? 그러면 LaunchnedEffect(key)도 key 값의 변경을 추적하는데, 그냥 LaunchedEffect랑은 무슨 차이인지 궁금해. 찾아보고 있긴 한데 아직 이해를 못 했어

```suggestion
    LaunchedEffect(pagerState) {
        snapshotFlow { pagerState.currentPage }
            .collect(viewModel::syncCurrentPage)
    }
```

다른 곳에서 사용한 snapshotFlow랑 조금 코드가 다르던데, 내 생각에는 이렇게 변경해야 할 것 같긴 한데 내가 틀릴 가능성 매우 높음! 
launchIn은 코루틴 스코프가 아닐 경우 스코프 지정해주는 terminal operator라서 만약 코루틴 스코프 내부라면 그냥 collect 해주면 돼!

Reply:
그르네 왜 일케했지